src/Repository/Provider/InMemoryRepositoryProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Repository\Provider;
+
+use Repository\Repository;
+
+/**
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+class InMemoryRepositoryProvider implements RepositoryProviderInterface
+{
+    private $repositories = array();
+
+    public function __construct(array $repositories)
+    {
+        foreach ($repositories as $repository) {
+            $this->addRepository($repository);
+        }
+    }
+
+    /**
+     * @param string $vendor
+     * @param string $name
+     *
+     * @return Repository|null
```

Comment:
I think it's better to throw an exception if the repository is not disabled. I can't imagine much use cases to return null.